scipy/sparse/data.py

```diff
@@ -297,7 +311,10 @@ def max(self, axis=None, out=None):
             Axis along which the sum is computed. The default is to
             compute the maximum over all the matrix elements, returning
             a scalar (i.e., `axis` = `None`).
-
+        explicit : {False, True} optional
+            Indicates that the nonzero entries of the matrix should be
+            considered. If a row/column is empty, a zero will be returned
+            to indicate it contains no nonzero values.
```

Comment:
> If a row/column is empty, a zero will be returned indicate it contains no nonzero values.

This is highly nonstandard behavior, no? `max([])` is an exception in Python and `csr_matrix((2,0)).max(axis=1)` is an exception in SciPy. I do not think returning 0 from max/min of an empty list is considered correct behavior.

Reply:
The row/column is empty only in the sense there are no explicit values contained in the row/column, so that in dense form it would only contain zeros. Returning a zero under those conditions is correct because only zeros are present in the row/column.

Reply:
Yes, but when `explicit=True`, the implicit zeros are not considered in the `min`/`max` operation. It is `min`/`max`ing over an empty list of explicit elements.

Reply:
Yes, but when `explicit=True`, the implicit zeros are not considered in the `min`/`max` operation. It is `min`/`max`ing over an empty list of explicit elements.

Reply:
Would returning a NaN be more appropriate?

Reply:
Ideally, `min`/`max` would return None on an empty vector, but you can't do that with a NumPy array. NaN is a fine sentinel for float32 and float64, but there is no equivalent for integer types.

Another possibility is `min([]) == inf` and `max([]) == -inf`, for float dtypes, and 2**32-1 and -2**32 for integer dtypes. But that is pretty ugly.

This is admittedly a hard problem when the data structure does not allow for missing integer values.